test/integration/comprehensive-transformer-test.spec.ts

```diff
@@ -192,6 +192,44 @@ describe('Comprehensive Transformer Test', () => {
           });
         });
       });
+      describe('calculateNeededToTransformToUnderlying', () => {
```

Comment:
omfg, it took me like 5' and several re-reads to understand this tests lmao. the namings. hahaha.